sled-agent/api/src/lib.rs

```diff
@@ -596,6 +597,15 @@ pub struct ArtifactCopyFromDepotBody {
     pub depot_base_url: String,
 }
 
+#[derive(Serialize, JsonSchema)]
+pub struct ArtifactCopyFromDepotResponse {}
+
+#[derive(Debug, Serialize, JsonSchema)]
+pub struct ArtifactPutResponse {
+    pub datasets: usize,
+    pub successful_writes: usize,
```

Comment:
This looks great as a response; I might add some docs indicating what these fields mean to the caller.

Maybe:
```suggestion
    /// The number of valid M.2 artifact datasets we found on the sled.
    /// There is typically one of these datasets for each functional M.2.
    pub datasets: usize,
    
    /// The number of valid writes to the M.2 artifact datasets. This should
    /// be less than or equal to the number of artifact datasets.
    pub successful_writes: usize,
```